src/ray/core_worker/store_provider/plasma_store_provider.cc

```diff
@@ -293,13 +276,14 @@ Status CoreWorkerPlasmaStoreProvider::Get(
                                          /*timeout_ms=*/0,
                                          // Mutable objects must be local before ray.get.
                                          results,
-                                         got_exception));
+                                         &got_exception));
   }
 
-  // If all objects were fetched already, return. Note that we always need to
-  // call UnblockIfNeeded() to cancel the get request.
-  if (remaining.empty() || *got_exception) {
-    return UnblockIfNeeded(raylet_ipc_client_, ctx);
+  // If all objects were fetched successfully or if any of the returned
+  // objects contain an exception, clean up the Get request in the raylet
+  // and early exit.
+  if (remaining.empty() || got_exception) {
+    return raylet_ipc_client_->CancelGetRequest();
```

Comment:
Note that there is a behavior difference here. This will now call `CancelGetRequest` from background threads in normal tasks where previously `NotifyWorkerUnblocked` was not called. This can exacerbate the "accidental cancelation" problem where one thread cancels outstanding get requests from others. I'm OK with this as an intermediate state since the next PR will presumably implement the get request ID logic.